### PR TITLE
Update retry guidelines

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

**Review by ahsonkhan — APPROVED** [review]
Should we broadcast this to all the languages to try to get compliance?

For example:
https://github.com/Azure/azure-sdk-for-java/blob/76cc5783df8321cc2a63c8f3b5033081b597f203/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/ExponentialBackoff.java#L26

File: docs/general/azurecore.md [hunk 1]
```diff
@@ -109,7 +109,9 @@ The HTTP Pipeline provides this functionality.
 
 {% include requirement/MUSTNOT id="azurecore-http-retry-requestid" %} change any client-side generated request-id when retrying the request. Th request ID represents the logical operation and should be the same across all physical retries of this operation.  When looking at server logs, multiple entries with the same client request ID show each retry
 
-{% include requirement/SHOULD id="azurecore-http-retry-defaults" %} implement a default policy that starts at 3 retries with a 0.8s delay with exponential (plus jitter) backoff.
+{% include requirement/SHOULD id="azurecore-http-retry-defaults" %} implement a default retry policy of 3 retries with a 0.8s exponential (plus jitter) delay between attempts with a maximum delay of 60s between attempts.
+
+{% include requirement/MUST id="azurecore-http-retry-defaults-overrides" %} allow the customer to override the default retry policy delay and timeouts.
```
Comment (ahsonkhan) [inline docs/general/azurecore.md:114]: From the offline discussion, we may want to add this as well:
> YOU SHOULD allow the customer to cancel the operation either via explicit cancellation or by setting an overall maximum operation time.

Comment (JeffreyRichter) [reply docs/general/azurecore.md:114]: Please add the YOU SHOLD about cancellation.

Comment (JeffreyRichter) [reply docs/general/azurecore.md:114]: You could broadcast the change but I think all architects were on the email thread so they should all know about it.

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**